File: python-project-euler-100/p019.py
import datetime
from Solution import Solution

class P019(Solution):
    def solve(self):
        self.problem_number = 19
        result = 0
        first_day = datetime.date(1901,1,1)
        delta = datetime.timedelta(days=5)
        first_sunday = first_day + delta
        delta = datetime.timedelta(days=7)
        current = first_sunday
        last_day = datetime.date(2000,12,31)
        while (last_day - current).days > 0:
            current = current + delta
            if (current.day == 1):
                result += 1
        return result

def main():
	P019().run()

if __name__ == "__main__":
	main()

File: typescript-project-euler-100/src/p019.ts
import { Solution } from "./Solution";

export class P019 extends Solution {

    solve() {
        this.problemNumber = 19;

		let result = 0;
		for (let year = 1901; year <= 2000; year++) {
			for (let month = 0; month < 12; month++) {
				const date = new Date(year, month, 1);
				if (date.getDay() === 0) {
					result++;
				}
			}
		}

		return result;
    }
}
